File: src/main/java/io/github/komdosh/alarms/actions/Alarm10Min.java
package io.github.komdosh.alarms.actions;

import com.intellij.openapi.actionSystem.AnAction;
import com.intellij.openapi.actionSystem.AnActionEvent;
import com.intellij.openapi.project.Project;
import io.github.komdosh.alarms.Alarms;

import java.util.concurrent.TimeUnit;

public class Alarm10Min extends AnAction {

    public Alarm10Min() {
        super("Alarm10Min");
    }

    public void actionPerformed(AnActionEvent event) {
        Project p = event.getProject();

        Alarms.setAlertInSeconds(p, TimeUnit.MINUTES.toSeconds(10), "Minutes");
    }
}

File: src/main/kotlin/io/github/komdosh/alarms/actions/Alarm10Min.kt
package io.github.komdosh.alarms.actions

import com.intellij.openapi.actionSystem.AnAction
import com.intellij.openapi.actionSystem.AnActionEvent
import com.intellij.openapi.components.ServiceManager
import io.github.komdosh.alarms.services.AlarmService
import java.util.concurrent.TimeUnit

class Alarm10Min : AnAction("Alarm10Min") {
    override fun actionPerformed(event: AnActionEvent) {
        event.project?.let{
            val alarmService = ServiceManager.getService(AlarmService::class.java)
            alarmService.setAlertInSeconds(it, TimeUnit.MINUTES, 10)
        }
    }
}
